drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -253,12 +261,26 @@ static int flash_npcx_nor_read(const struct device *dev, off_t addr,
 	return 0;
 }
 
-static int flash_npcx_nor_erase(const struct device *dev, off_t addr,
-				  size_t size)
+static void disable_multiple_erase_sizes(void)
+{
+#undef ERASE_BLOCK_64K
+#undef ERASE_BLOCK_4K
+}
+
+static void enable_multiple_erase_sizes_within_range(const struct flash_npcx_nor_config *config)
+{
+	if (config->min_erase_size <= BLOCK_64K_SIZE && BLOCK_64K_SIZE <= config->max_erase_size) {
+#define ERASE_BLOCK_64K
```

Comment:
this line always define ERASE_BLOCK_4K...
I believe you should use a variable for runtime code flow control

C Preprocessor is build time thing.